data/adapters/file_fileset_adapter.py

```diff
@@ -0,0 +1,228 @@
+import requests
+import json
+
+from typing import Optional
+from adapters.writer import Writer
+
+
+class FileFileSet:
+    ALLOWED_LABELS = ['encode_file_fileset']
+
+    def __init__(
+        self,
+        accession: str,
+        label='encode_file_fileset',
+        writer: Optional[Writer] = None,
+        **kwargs
+    ):
+        if label not in FileFileSet.ALLOWED_LABELS:
+            raise ValueError('Invalid label. Allowed values: ' +
+                             ','.join(FileFileSet.ALLOWED_LABELS))
+
+        self.label = label
+        self.writer = writer
+        self.accession = accession
+
+    def process_file(self):
+        self.writer.open()
+        if self.label == 'encode_file_fileset':
+            _props = query_fileset_files_props_encode(self.accession)
+            self.writer.write(json.dumps(_props) + '\n')
+
+        self.writer.close()
+
+
+def query_fileset_files_props_encode(accession):
```

Comment:
I think we should break down this method so it's easier to read/modify later, maybe `parse_prediction()`, `parse_sample_and_treatment()`, ... returning parts of the final json